### ISSUE TITLE
update angular version for jupyter-apis

### ISSUE THREAD

**Issue by mathis-marcotte** [StatCan/jupyter-apis#306]
Opened this issue upstream in the past https://github.com/kubeflow/kubeflow/issues/7172. 

It hasn't gotten picked up yet, and the latest update was a request for help on doing the work. So I figured might as well do this work ourselves now.

**Comment by mathis-marcotte** [comment]
I will be using [this tool](https://update.angular.io/?l=2&v=12.0-17.0) to go through the migration process. I will probably try to upgrade one version at a time to make sure the debugging is easier to do if any issues arise.

**Comment by mathis-marcotte** [comment]
While working on this, I'll remove all the package overrides we were doing to fix vulnerabilities. With the update to some of our large packages (mainly talking about the angular-devkit package that seemed to have a lot of those vulnerabilities), those overrides might end up not being necessary. I'll re-assess if those overrides are needed once I am done updating angular.

**Comment by mathis-marcotte** [comment]
Upgraded from angular v12 to v13. All cypress tests pass
![image](https://github.com/StatCan/jupyter-apis/assets/84033116/b4ae0be6-e2b0-46b3-88a6-6df63788135e)

**Comment by mathis-marcotte** [comment]
After upgrading to v14, things looked okay when I tested locally, but some of the cypress tests are failing, so I'll look into why that is. 
![image](https://github.com/StatCan/jupyter-apis/assets/84033116/fdb1da93-672c-4b85-bebe-438bc96dd01a)

**Comment by mathis-marcotte** [comment]
I'm able to solve that issue I last talked about by deleting the folder ".angular", "node_modules" and "dist" that can be found under the `kubeflow-common-lib` folder and the` jupyter` folder (these are all folders that we gitignore). After doing that and doing a `npm ci` command instead of the usual `npm i` install command, that solved it and the app was now using the correctly defined namespace. So after than, I could successfully run all the cypress tests.
![image](https://github.com/StatCan/jupyter-apis/assets/84033116/1f3531eb-91a6-4ca9-b408-f03d70031b74)

With that done, angular v14 seems to be working correctly so I will move on to trying to install v15.

**Comment by mathis-marcotte** [comment]
Major changes to upgrade to angular 17 is node going from v16 to v18, and typescript going from v4 to v5. Will have to make sure the apps work after those updates. And will have to make sure to use node v18 from now on.

**Comment by mathis-marcotte** [comment]
I can't currently update @angular/material to v17. It does not contain the legacy components that we are currently using. v16 can still work for that package for now while I update the other angular packages to v17. But now it adds a bit more pressure on needing to update those legacy components.

**Comment by mathis-marcotte** [comment]
Updated to angular17 (except for @angular/material, because of the reason stated in the comment above).
Everything seems to work as expected. Cypress tests all pass
![image](https://github.com/StatCan/jupyter-apis/assets/84033116/94936188-5ae2-414c-96e0-0cefaed0c5fb)

Before merging, I'll also probably want to take a quick look at the remaining vulnerabilities that we are getting from npm. I removed the package version overrides we were doing before starting to upgrade angular in case some of those got fixed in the more recent angular versions(some of which definitely looked that way).

Also, THIS SHOULD BE MERGED FOR AAW2.0!
